components/automate-deployment/pkg/target/local_target.go

```diff
@@ -334,25 +334,29 @@ func (t *LocalTarget) SetUserToml(name, config string) error {
 
 // InstallService installs an automate service. Returns an error if
 // the install failed for any reason.
-func (t *LocalTarget) InstallService(svc habpkg.Installable, channel string) error {
-	return t.installPackage(svc, channel)
+func (t *LocalTarget) InstallService(ctx context.Context, svc habpkg.Installable, channel string) error {
+	return t.installPackage(ctx, svc, channel)
 }
 
 // unloadServiceWithHabVersion unloads the given service with a
 // particular version of hab and hab-sup. This is used during the
 // Habitat upgrade process to unload services before upgrading hab.
-func (t *LocalTarget) unloadServiceWithHabVersion(svc habpkg.VersionedPackage, binPkg habpkg.HabPkg, habSupPkg habpkg.HabPkg) error {
+func (t *LocalTarget) unloadServiceWithHabVersion(ctx context.Context, svc habpkg.VersionedPackage, binPkg habpkg.HabPkg, habSupPkg habpkg.HabPkg) error {
```

Comment:
ummmm, any idea if it's possible to get rid of this? If not, we maybe should take your fix from above?

Reply:
Good catch! Whether or not we need to worry about unloading with a specific hab/hab-sup I can't speak to. As per using the same implementation under the hood, not really unless we either add a new function to the HabCmd interface or modify the existing one to require specifying them. I think the former is probably better long term, but for now I think just patching both functions is the least intrusive until we want to do another refactor.

Reply:
I believed that to get rid of this we need to stop supporting upgrades from old versions of deployment-service. Namely, this is used so that we can unload services from hab-sup even if the running hab-sup is very old.